api/server/httputils/errors.go

```diff
@@ -54,7 +54,7 @@ func GetHTTPErrorStatusCode(err error) int {
 			code    int
 		}{
 			{"not found", http.StatusNotFound},
-			{"no such", http.StatusNotFound},
+			{"no such", http.StatusBadRequest},
```

Comment:
Maybe we can add `{"invalid reference format", httphttp.StatusBadRequest},` before `{"no such", http.StatusNotFound}` @thaJeztah